engine/hud/ClickableHudElement.cpp

```diff
@@ -21,6 +21,7 @@ bool ClickableHudElement::isCoordOnElement(int x, int y) {
 void ClickableHudElement::onInput(SDL_Event *event) {
     // If its a mouse button up event, it cannot be clicked
     if (mIsClicked && event->type == SDL_MOUSEBUTTONUP) {
+        SDL_StartTextInput();
```

Comment:
Strongly suggest moving this into input field since it is the only clickable hud element that will accept text input. You can override update() and call the base update after handling text.